signetdev/host/signetdev.c

```diff
@@ -52,9 +53,15 @@ int signetdev_device_num_root_blocks()
 	return 1;
 }
 
-int signetdev_priv_cmd_payload_size()
+unsigned int signetdev_priv_cmd_payload_size()
 {
-	return signetdev_device_block_size() + CMD_PACKET_HEADER_SIZE + 1;
+    int bs = signetdev_device_block_size();
+    if (bs == -1) {
```

Comment:
I think the right solution here is for signetdev_device_block_size() to also be an unsigned int so we can skip this checking

Reply:
`signetdev_device_block_size()` used -1 to signal error but then here this is possibly masked by `+ CMD_PACKET_HEADER_SIZE + 1`

Of course, `signetdev_device_block_size()` may return `0` as error, but we still should check for that, not just add numbers to it.

Reply:
See my latest comment on the PR. Lets defer this problem and not add any new error handling, at least not immediately